src/Compilers/Shared/BuildClient.cs

```diff
@@ -219,6 +217,13 @@ private int RunLocalCompilation(string[] arguments, BuildPaths buildPaths, TextW
                 return null;
             }
 
+            // A temporary directory is required for the server to startup hence we bail out early
+            // here if it's not available.
+            if (buildPaths.TempDirectory is null)
+            {
+                return null;
```

Comment:
>return null; [](http://example.com/codeflow?start=16&length=12)

Should we write a message to the log here?

Reply:
yeah we should.